src/interactivity/win32/uiaTextRange.cpp

```diff
@@ -0,0 +1,278 @@
+                                   const UiaPoint point)
+{
+    UiaTextRange* range = nullptr;
+    try
+    {
+        range = new UiaTextRange(pData, pProvider, point);
+    }
+    catch (...)
+    {
+        range = nullptr;
+    }
+
+    if (range)
+    {
+        pProvider->AddRef();
+    }
+    return range;
+}
+
+// degenerate range constructor.
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData, _In_ IRawElementProviderSimple* const pProvider) :
+    UiaTextRangeBase(pData, pProvider)
+{
+}
+
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData,
+                           _In_ IRawElementProviderSimple* const pProvider,
+                           const Cursor& cursor) :
+    UiaTextRangeBase(pData, pProvider, cursor)
+{
+}
+
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData,
+                           _In_ IRawElementProviderSimple* const pProvider,
+                           const Endpoint start,
+                           const Endpoint end,
+                           const bool degenerate) :
+    UiaTextRangeBase(pData, pProvider, start, end, degenerate)
+{
+}
+
+// returns a degenerate text range of the start of the row closest to the y value of point
+UiaTextRange::UiaTextRange(_In_ Microsoft::Console::Render::IRenderData* pData,
+                           _In_ IRawElementProviderSimple* const pProvider,
+                           const UiaPoint point) :
+    UiaTextRangeBase(pData, pProvider)
+{
+    Initialize(point);
+}
+
+IFACEMETHODIMP UiaTextRange::Clone(_Outptr_result_maybenull_ ITextRangeProvider** ppRetVal)
+{
+    try
+    {
+        *ppRetVal = new UiaTextRange(*this);
+    }
+    catch (...)
+    {
+        *ppRetVal = nullptr;
+        return wil::ResultFromCaughtException();
+    }
+    if (*ppRetVal == nullptr)
+    {
+        return E_OUTOFMEMORY;
+    }
+
+#if defined(_DEBUG) && defined(UiaTextRangeBase_DEBUG_MSGS)
+    OutputDebugString(L"Clone\n");
+    std::wstringstream ss;
+    ss << _id << L" cloned to " << (static_cast<UiaTextRangeBase*>(*ppRetVal))->_id;
+    std::wstring str = ss.str();
+    OutputDebugString(str.c_str());
+    OutputDebugString(L"\n");
+#endif
+    // TODO GitHub #1914: Re-attach Tracing to UIA Tree
+    // tracing
+    /*ApiMsgClone apiMsg;
+    apiMsg.CloneId = static_cast<UiaTextRangeBase*>(*ppRetVal)->GetId();
+    Tracing::s_TraceUia(this, ApiCall::Clone, &apiMsg);*/
+
+    return S_OK;
+}
+
+IFACEMETHODIMP UiaTextRange::FindText(_In_ BSTR text,
+                                      _In_ BOOL searchBackward,
+                                      _In_ BOOL ignoreCase,
+                                      _Outptr_result_maybenull_ ITextRangeProvider** ppRetVal)
+{
+    // TODO GitHub #1914: Re-attach Tracing to UIA Tree
+    //Tracing::s_TraceUia(this, ApiCall::FindText, nullptr);
+
+    *ppRetVal = nullptr;

```

Comment:
This file has a bunch of them too. We're setting *ppRetVal without checking if ppRetVal was valid (should be a RETURN_HR_IF(E_INVALIDARG, ...) before attempting to set)